### PR TITLE
dird: remove optimize_for_size/optimize_for_speed

### PR DESCRIPTION
**Pull request by sebsura**
### Thank you for contributing to the Bareos Project!

This PR removes the `optimize_for_*` options from the director.  It also fixes a memory leak.
It also includes a fix for a memory leak inside the director.

The backport to 23 will only contain the deprecation of these options.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Required backport PRs have been created

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

### CODE REVIEW

File: core/src/dird/ua_restore.cc [hunk 1]
```diff
@@ -335,6 +335,16 @@ bool RestoreCmd(UaContext* ua, const char*)
 
   if (regexp) { free(regexp); }
 
+  /* restore_tree_root only gets freed if either the backup starts
+   * or the ua session ends.  Since the first definitely does not happen
+   * after this point, and we cannot control the second one, we need
+   * to ensure that we free the tree here before returning false; otherwies
```
Comment (pstorz) [inline core/src/dird/ua_restore.cc:341]:
```suggestion
   * to ensure that we free the tree here before returning false; otherwise
```

**Review by pstorz — CHANGES_REQUESTED** [review]
Looks good! Only one typo in the comment.
Backport needs only to deprecate the directives.
Good work!

File: core/src/dird/ua_restore.cc (hunk 1 above)
Comment (sebsura) [reply core/src/dird/ua_restore.cc]: fixed

**Review by pstorz — CHANGES_REQUESTED** [review]
Looks good, but why do we have so many empty files?(.rst.inc).
I guess we can leave them away?

**Review by pstorz — APPROVED** [review]

**Pull request merged**